pytest_fixtures/core/contenthosts.py

```diff
@@ -107,21 +107,45 @@ def registered_hosts(organization_ak_setup, content_hosts, default_sat):
 @pytest.fixture(scope="function")
 def katello_host_tools_host(default_sat, module_org, rhel_contenthost):
     """Register content host to Satellite and install katello-host-tools on the host."""
+    repo = settings.repos['SATCLIENT_REPO'][f'RHEL{rhel_contenthost.os_version.major}']
+    register_host_custom_repo(default_sat, module_org, rhel_contenthost, [repo])
+    rhel_contenthost.install_katello_host_tools()
+    yield rhel_contenthost
+
+
+@pytest.fixture(scope="function")
+def cockpit_host(default_sat, module_org, rhel_contenthost):
+    """Register content host to Satellite and install cockpit on the host."""
+    rhelver = rhel_contenthost.os_version.major
+    if rhelver > 7:
+        repo = [settings.repos[f'rhel{rhelver}_os']['baseos']]
+    else:
+        repo = [settings.repos['rhel7_os'], settings.repos['rhel7_extras']]
+    register_host_custom_repo(default_sat, module_org, rhel_contenthost, repo)
+    rhel_contenthost.execute(f"hostnamectl set-hostname {rhel_contenthost.hostname} --static")
```

Comment:
just curious, why this is needed? Isn't this pointing to some problem at satlab side?

Reply:
The cockpit is using static hostname in UI, so without this command, it is reporting localhost. Not sure if we want to set the static hostname on all hosts from satlab. Without this command transient hostname is still set to the correct value.